### ISSUE TITLE
docs: lead searches.py guide with Nautilus, MultiStartProdigy second

### ISSUE THREAD

**Issue by Jammy2211** [Jammy2211/autolens_workspace#386]
## Overview

`scripts/guides/modeling/searches.py` is the workspace's tour of every non-linear search available for modeling. In **both** `autolens_workspace` and `autogalaxy_workspace` it currently opens on **Dynesty** and never documents **Nautilus at all** — despite naming Nautilus repeatedly, in its own intro and in the Dynesty / Zeus / LBFGS sections, as the recommended search. A reader going top-to-bottom meets four searches they are told not to use before reaching either of the two that are actually recommended.

The guide has drifted from what the scripts do today. `imaging/modeling.py` fits with `af.Nautilus` and `imaging/start_here.py` fits with `af.MultiStartProdigy` in both repos — yet autogalaxy's guide still documents the older `MultiStartAdam` with a hand-tuned `learning_rate`, and both intros still claim Nautilus is used in every modeling example.

This task makes both files lead with **Nautilus**, then **MultiStartProdigy**, with the remaining searches following as the alternatives they are.

## Plan

- Add a new `__Nautilus__` section as the **first** search in both files: full posterior sampling (errors + covariances), gradient-free so it does **not** use JAX gradients, but **does** exploit the JAX GPU via batched `n_batch` likelihood evaluation, plus the `n_live` accuracy/run-time trade-off and a runnable cell.
- Move the multi-start gradient-optimizer section to **second**, directly after Nautilus and ahead of Dynesty/Emcee/Zeus/LBFGS.
- In autogalaxy, retitle that section `__MultiStartProdigy__` and switch the cell to `af.MultiStartProdigy` (dropping `learning_rate`), matching that repo's own `start_here.py`. Keep it short-form — do not port autolens's pixelized-source subsection.
- Fix the cross-references the reorder breaks (the multi-start prose refers to LBFGS as "noted above", which stops being true once it sits above LBFGS).
- Reorder the `__Contents__` list with a Nautilus bullet, and reframe the intro as a two-search recommendation: Nautilus when you need errors, MultiStartProdigy when you want a fast check.

<details>
<summary>Detailed implementation plan</summary>

### Affected Repositories
- `autolens_workspace` (primary)
- `autogalaxy_workspace`

### Branch Survey

| Repository | Current Branch | Dirty? |
|-----------|---------------|--------|
| ./autolens_workspace | main | clean |
| ./autogalaxy_workspace | main | clean |

**Suggested branch:** `feature/searches-guide-nautilus-first`

**Soft conflict, hand-verified — not a block.** Both repos are claimed in `active.md` by two other tasks, `remove-finish-docstring-hack` and `script-prose-ref-drift`, both awaiting-merge with open PRs. Neither touches `guides/modeling/searches.py` **or** `searches.ipynb` in either repo (verified via `git diff --name-only origin/main...origin/<branch>`), so the file sets are disjoint and a third worktree off `main` is safe.

**Brain override (recorded):** `pyauto-brain feature` scored this `too-large (10)` and proposed a 4-phase split (`design` / `core_api` / `workspace_examples` / `docs`). That is the known repo-count difficulty proxy misfiring — this is two prose files with zero library API change, so a `core_api` phase is vacuous. Overridden to **small, single pair of PRs**.

### Target section order (both files)

```
Nautilus → MultiStartProdigy → Dynesty → Emcee → Zeus → LBFGS → Start Point → Search Cookbook
```

Currently: `Dynesty → Emcee → Zeus → LBFGS → (multi-start) → Start Point → Search Cookbook`.

### Implementation Steps

**1. New `__Nautilus__` section, first in both files.** Written fresh — there is no existing block to move. Anchored on the prose and settings already in each repo's `imaging/modeling.py`. Four beats:

- **Full posterior.** Nautilus does not return a single best-fit model; it maps the full posterior — every parameter's probability density, its errors, and the covariances between parameters. That is why it stays the default recommendation even though the optimizer below is faster: an optimizer gives you a point, Nautilus gives you a number you can quote.
- **Gradient-free — no JAX gradients.** It explores by drawing new live points from a learned boundary around the current live set, using likelihood evaluations only. It never differentiates the likelihood, so unlike `MultiStartProdigy` it neither needs nor uses JAX's gradients.
- **But it does exploit the JAX GPU.** Verified against the installed `autofit`, not assumed: `autofit/non_linear/search/nest/nautilus/search.py:196-199` builds the `Fitness` with `use_jax_vmap=True` and `batch_size=self.n_batch` whenever the analysis has `_use_jax`, and passes `vectorized=fitness.use_jax_vmap` to the sampler (`:291`); `autofit/non_linear/fitness.py:509` is `jax.vmap(jax.jit(self.call))`. So Nautilus proposes `n_batch` points per step and all of them are fitted **simultaneously in one GPU call**. The JAX speed-up applies in full — it just comes from batched likelihood evaluation rather than gradient descent. `n_batch` is also the main VRAM control.
- **`n_live`.** The accuracy/run-time trade-off: more live points are more accurate and slower, fewer are faster but risk a local maxima; complex models need more. No specific "default" value is quoted — the `autofit` signature default is 3000 while `modeling.py`'s prose says 200, and reconciling that is out of scope.

Followed by a runnable cell matching `modeling.py`:

```python
search = af.Nautilus(
    path_prefix=Path("imaging", "searches"),
    name="Nautilus",
    unique_tag="example",
    n_live=100,   # increase for more complex models
    n_batch=50,   # model fits are batched and run simultaneously on the GPU
    iterations_per_quick_update=10000,
)
```

**2. Move the multi-start section second.** Relocate the existing block to sit directly after Nautilus, ahead of Dynesty.

- *autolens:* move as-is — it already reads `__MultiStartProdigy__` with the GIGA-Lens and Prodigy citations and the `__Pixelized sources__` subsection. Content preserved verbatim except for the cross-reference fix in step 4.
- *autogalaxy:* retitle `__MultiStartAdam__` → `__MultiStartProdigy__` and switch the cell to `af.MultiStartProdigy` (dropping `learning_rate=0.01`). Keep autogalaxy's concise register — list `MultiStartAdam` / `MultiStartADABelief` as alternatives; **do not** port the pixelized-source subsection.

```python
search = af.MultiStartProdigy(
    path_prefix=Path("imaging", "searches"),
    name="MultiStartProdigy",
    n_starts=50,
    n_steps=500,
)
```

**3. Reframe the intro (both files).** It currently reads "the default search used throughout all modeling examples, `Nautilus`" and "unless you really know what you are doing … stick to Nautilus". Rewrite as a two-search recommendation reflecting what the scripts actually do: Nautilus when you need errors and a posterior (`modeling.py`), MultiStartProdigy when you want a fast check that the model and data are sensible (`start_here.py`); everything below is an alternative. Keep the existing nested-sampler / MCMC / MLE categorisation paragraph.

**4. Fix the cross-references the reorder breaks.**

- The multi-start prose says it addresses the weakness of a single-start optimizer "like `LBFGS` **noted above**" — false once it sits above LBFGS → "described below".
- Sweep both files for any other positional wording invalidated by the move. `Emcee → Zeus` ordering is unchanged, so "`Zeus`, described next" is fine; "does not use the start-point API described below" stays correct.
- Update the `__Contents__` list: add a Nautilus bullet and reorder to match.

### Out of scope

- The `path_prefix=Path("searches")` vs `Path("imaging", "searches")` inconsistency already present between the Dynesty cell and every other cell in the file.
- The stale `n_live` default claim in `imaging/modeling.py`.
- Any other guide, `start_here.py`, or `modeling.py`.

### Key Files

- `autolens_workspace/scripts/guides/modeling/searches.py` — the guide; add Nautilus, move MultiStartProdigy second, fix the LBFGS cross-reference, reorder Contents, reframe intro.
- `autogalaxy_workspace/scripts/guides/modeling/searches.py` — same, plus retitle `MultiStartAdam` → `MultiStartProdigy` and switch the cell.
- `*/notebooks/guides/modeling/searches.ipynb` — regenerated via PyAutoHands `generate.py`, never hand-edited.
- Reference for prose/settings (read, not edited): `*/scripts/imaging/modeling.py` (Nautilus), `*/scripts/imaging/start_here.py` (MultiStartProdigy).

### Verification

1. `python .github/scripts/run_smoke.py` in each workspace — `searches.py` is a guide, so confirm whether it is listed in `smoke_tests.txt`; if not, execute it directly under the guide's declared `ENV: full_datasets` profile.
2. Grep both files for `noted above` / `described next` / `described below` to confirm no stale positional reference survives.
3. `af.Nautilus` / `af.MultiStartProdigy` kwargs — already checked against the installed `autofit`: both exist; `n_live`, `n_batch`, `iterations_per_quick_update` are real `Nautilus` parameters.
4. Regenerate notebooks and confirm only `notebooks/guides/modeling/searches.ipynb` changes.
5. `scripts/check_sizes.sh` in each workspace — both files should **grow**, so no shrink flag.

</details>

## Original Prompt

<details>
<summary>Click to expand starting prompt</summary>

> For searches.py in autolens and galaxy workspace, can you move a section on Nautilus up to the top, explaining its full posterior sasmpling but does not use JAX gradients (but does exploit JAX GPU). Then put MultiStartProdgy second.

</details>

**Comment by Jammy2211** [comment]
## Shipped — both PRs open

- **autolens_workspace** → PyAutoLabs/autolens_workspace#388 (`81375671`)
- **autogalaxy_workspace** → PyAutoLabs/autogalaxy_workspace#188 (`d6f5e31`)

Both branched off post-merge `main` (autolens `264d4ab9`, autogalaxy `7c5e43b`), both labelled `pending-release`.

### What landed

New section order in both guides:

```
Nautilus → MultiStartProdigy → Dynesty → Emcee → Zeus → LBFGS → Start Point → Search Cookbook
```

**The Nautilus section was written fresh, not moved** — neither file documented Nautilus at all, despite naming it throughout as the recommended search. It covers the full posterior (errors + covariances), the fact that it is gradient-free and so does not use JAX gradients, and the fact that it nonetheless exploits the JAX GPU: `n_batch` proposals are evaluated simultaneously through the `vmap`/`jit`-wrapped likelihood.

That GPU claim was verified against the installed `autofit` rather than assumed — `search/nest/nautilus/search.py:196-199` builds the `Fitness` with `use_jax_vmap=True` and `batch_size=self.n_batch` whenever the analysis has `_use_jax`, passes `vectorized=fitness.use_jax_vmap` to the sampler (`:291`), and `fitness.py:509` is `jax.vmap(jax.jit(self.call))`.

**autogalaxy carried an extra fix:** its section documented the older `MultiStartAdam` with a hand-tuned `learning_rate=0.01`, while that repo's own `imaging/start_here.py` already fits with `af.MultiStartProdigy`. Retitled and switched, `learning_rate` dropped. Kept short-form — autolens's pixelized-source subsection was deliberately not ported.

Also fixed the cross-reference the reorder invalidated (the multi-start prose called LBFGS's weakness "noted above", now "described below"), reordered `__Contents__`, and reframed both intros as the two-search recommendation the scripts actually follow.

### Gate + validation

- **Heart YELLOW, acknowledged by the human.** Reasons recorded verbatim in `active.md`: `workspace validation not passing (0 failed, cloud#30516167217)` and `release validation stale: source moved since rehearsal (PyAutoFit, PyAutoArray, PyAutoGalaxy, PyAutoLens)`. `red_reasons: []`, score 70. Neither relates to this change.
- Both scripts executed directly (the guide is not in `smoke_tests.txt`) — exit 0 in each workspace.
- `check_sizes.sh` — `OK: all scripts within size tolerance` in both.
- Notebooks + catalogue regenerated via PyAutoHands; the diff is confined to `guides/modeling/searches` entries.
- CI: `navigator` and `Catalogue staleness` pass on both; `smoke` in flight.

### Notes

Brain scored this `too-large (10)` and proposed a four-phase split (`design`/`core_api`/`workspace_examples`/`docs`) — the known repo-count difficulty proxy misfiring. Overridden to a single pair of PRs: two prose files, zero library API change, so a `core_api` phase was vacuous.

Awaiting merge — merge and issue closure remain human calls.

**Comment by Jammy2211** [comment]
## Merged — closing

- **autolens_workspace** → PyAutoLabs/autolens_workspace#388 MERGED (`f5d03984`)
- **autogalaxy_workspace** → PyAutoLabs/autogalaxy_workspace#188 MERGED (`78aff831`)

All 16 CI checks passed on both before merge — `smoke (3.12)`, `smoke (3.13)`, `navigator / paths + banner lint`, `navigator / Catalogue staleness`. The catalogue-staleness pass confirms the regenerated `llms-full.txt` / `workspace_index.json` match the edited scripts.

Both guides now read `Nautilus → MultiStartProdigy → Dynesty → Emcee → Zeus → LBFGS → Start Point → Search Cookbook`.
